cluster/addons/registry/README.md

```diff
@@ -0,0 +1,256 @@
+# Private Docker Registry in Kubernetes
+
+Kubernetes offers an optional private Docker registry addon, which you can turn
+on when you bring up a cluster or install later.  This gives you a place to
+store truly private Docker images for your cluster.
+
+## How it works
+
+The private registry runs as a `Pod` in your cluster.  It does not currently
+support SSL or authentication, which triggers Docker's "insecure registry"
+logic.  To work around this, we run a proxy on each node in the cluster,
+exposing a port onto the node (via a hostPort), which Docker accepts as
+"secure", sicne it is accessed by `localhost`.
+
+## Turning it on
+
+Some cluster installs (e.g. GCE) support this as a cluster-birth flag.  The
+`ENABLE_CLUSTER_REGISTRY` flag in `cluster/gce/config-default.sh` governs
```

Comment:
ENABLE_CLUSTER_REGISTRY is used internally. If a user wanted to enable the cluster they would want to set KUBE_ENABLE_CLUSTER_REGISTRY=true.